### PR TITLE
MakeSSP

### PR DESCRIPTION
**Pull request by Tom-Camp**
Poetryizing MakeSSP and a little restructuring.

### CODE REVIEW

File: tools/makessp/makessp.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
+"""
+Copyright 2019-2020 CivicActions, Inc. See the README file at the top-level
```
Comment (openprivacy) [inline tools/makessp/makessp.py]: "Copyright 2019-2024"

File: tools/helpers/ssp.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+"""
+Copyright 2019-2020 CivicActions, Inc. See the README file at the top-level
```
Comment (openprivacy) [inline tools/helpers/ssp.py]: "Copyright 2019-2024"

**Review by openprivacy — CHANGES_REQUESTED** [review]
Looks and works great - good job on the TOC!
Two small changes requested (Copyright comment).

File: tools/makessp/makessp.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        "",
+        f"# {project.name} System Security Plan",
+        "",
+    ]
+    standards = get_standards()
+    for standard in standards:
+        text_output.append(f"## {standard}")
+    text_output.append("\n")
+
+    for family in ssp_data.families:
```
Comment (openprivacy) [inline tools/makessp/makessp.py:72]: Couple issues just found:
1. Family order: AT and CA appear near the bottom of the TOC
2. Standard: outputs "NIST SP 800-53 Revision 5" but all controls are still "Revision 4"
3. If it were Revision 5, then we should see PT and SR families...
4. Also, controls.md (created by makefamilies) is mode 644 but ssp.md is mode 600

**Review by openprivacy — CHANGES_REQUESTED** [review]
Is makessp supposed to be complete?

File: tools/makessp/makessp.py (hunk 3 above)
Comment (Tom-Camp) [reply tools/makessp/makessp.py:72]: Do we have components that address PT or SR family controls?

Comment (Tom-Camp) [reply tools/makessp/makessp.py:72]: The file perm change is coming from the `md-toc` library. I commented out the toc call and the perms were 664, after the toc write they were 600. I'm wondering if this has always been there but we'd not noticed. I'll put in an issue with the md-toc maintainer.

Comment (openprivacy) [reply tools/makessp/makessp.py:72]: You're right that we don't have components with PT and SR families represented - yet.
Though we may want to allow choice of standard & certification as has been done on client site.

Comment (openprivacy) [reply tools/makessp/makessp.py:72]: That will be another commit. I'll approve this PR now - great work!

**Review by openprivacy — APPROVED** [review]

**Pull request merged**